code/modules/power/pt_laser.dm

```diff
@@ -243,13 +241,17 @@
 	if(online) // if it's switched on
 		if(!firing) //not firing
 
-			if(charge >= adj_output && (adj_output >= PTLMINOUTPUT)) //have power to fire
+			if((charge + input_power) >= adj_output && (adj_output >= PTLMINOUTPUT)) //have power to fire
 				start_firing() //creates all the laser objects then activates the right ones
 				dont_update = 1 //so the firing animation runs
-				charge -= adj_output
-		else if(charge < adj_output && (adj_output >= PTLMINOUTPUT)) //firing but not enough charge to sustain
+				input_power -= adj_output
+				if (input_power < 0)
+					charge += input_power
+		else if(((charge + (src.charging * input_power)) < adj_output) && (adj_output >= PTLMINOUTPUT)) //firing but not enough charge to sustain
```

Comment:
This is good, but I think it should drain the remaining charge in this case to indicate *why* it's shutting off. Also can this statement be replaced with `can_fire`, incorporating the `PTLMINOUTPUT` check into that if needed?